packages/client/src/runtime/getPrismaClient.ts

```diff
@@ -339,12 +340,17 @@ export function getPrismaClient(config: GetPrismaClientConfig) {
     _rejectOnNotFound?: InstanceRejectOnNotFound
     _dataProxy: boolean
     _extensions: Extension[]
+    _logEmitter: EventEmitter
 
     constructor(optionsArg?: PrismaClientOptions) {
       if (optionsArg) {
         validatePrismaClientOptions(optionsArg, config.datasourceNames)
       }
 
+      this._logEmitter = new EventEmitter()
```

Comment:
Different engines have their own instance of the EventEmitter, that for some reason is not a singleton. I thought it made more sense to inject the dependency, as the event pub/sub should be the same across the client codebase.

Then, wherever else it is used, it's injected.